app/components/DropdownMenu/DropdownMenu.js

```diff
@@ -26,9 +26,19 @@ type Props = {
 
 @observer
 class DropdownMenu extends React.Component<Props> {
-  @observable top: number;
-  @observable right: number;
-  @observable left: number;
+  @observable top: ?number;
+  @observable bottom: ?number;
+  @observable right: ?number;
+  @observable left: ?number;
+  @observable position: 'left' | 'right' | 'center';
+  @observable bodyRect: ClientRect;
+  @observable labelRect: ClientRect;
+  @observable dropdownRef: { current: null | HTMLElement };
+
+  constructor(props: Props) {
+    super(props);
+    this.dropdownRef = React.createRef();
```

Comment:
There's no need for the constructor here, you can actually just do this on the class itself

```javascript
dropdownRef: React.createRef();
```

Reply:
fixed, thanks!